### PR TITLE
8268457: XML Transformer outputs Unicode supplementary character incorrectly to HTML

### PR DESCRIPTION
**Pull request by masyano**
Hi all,

Could you please review the 8268457 bug fixes?

The problem is that ToHTMLStream applies processing for non-surrogate pairs to the surrogate pair.
This fix changes the processing for non-surrogate pairs to the else condition.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8268457](https://bugs.openjdk.java.net/browse/JDK-8268457): XML Transformer outputs Unicode supplementary character incorrectly to HTML

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)
 * [Joe Wang](https://openjdk.java.net/census#joehw) (@JoeWang-Java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4474/head:pull/4474` \
`$ git checkout pull/4474`

Update a local copy of the PR: \
`$ git checkout pull/4474` \
`$ git pull https://git.openjdk.java.net/jdk pull/4474/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4474`

View PR using the GUI difftool: \
`$ git pr show -t 4474`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4474.diff">https://git.openjdk.java.net/jdk/pull/4474.diff</a>

</details>

### CODE REVIEW

**Comment by JoeWang-Java** [comment]
The fix looks good to me.

For ToHTMLStream, please update the copyright year to 2021 (s/2019/2021) and s/@LastModified: Aug 2019/ @LastModified: June 2021.

For the test, please move it to test/jaxp/javax/xml/jaxp/unittest/transform and make it a testng test (add @run testng transform.SurrogateTest, and @Test to each test case). 
For the test itself, you may replace the try-finally block with a try-with-resources statement. For comparing with the gold files, please take a look at the existing test library, specifically test/jaxp/javax/xml/jaxp/libs/jaxp/library/JAXPTestUtilities.java, and do it the testng way, e.g. Assert.assertTrue(compareWithGold(goldFile, outputFile));

**Comment by masyano** [comment]
Thank you for reviewing. I reflected the review comments to my change.

File: src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/ToHTMLStream.java [hunk 1]
```diff
@@ -1451,22 +1451,24 @@ else if ((ch < m_maxCharacter) && (m_maxCharacter == 0xFFFF)
                     {
                     writer.write(ch);  // no escaping in this case
                     }
-                    else
```
Comment (naotoj) [inline src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/ToHTMLStream.java:1454]: I was suggesting removing the entire comment-out block if it is not needed (and confusing), but I will defer the decision to Joe.

File: test/jaxp/javax/xml/jaxp/unittest/transform/SurrogateTest1.xml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<root>
+    <tag1>𠮟</tag1>
+</root>
```
Comment (naotoj) [inline test/jaxp/javax/xml/jaxp/unittest/transform/SurrogateTest1.xml:4]: Add a new line at the end of the file (and other newly created files too).

File: src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/ToHTMLStream.java (hunk 1 above)
Comment (JoeWang-Java) [reply src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/ToHTMLStream.java:1454]: I agree. It's very obsolete. The comment-out block from line 1445 to 1454 can be removed.

Comment (masyano) [reply src/java.xml/share/classes/com/sun/org/apache/xml/internal/serializer/ToHTMLStream.java:1454]: I was mistaken. I deleted the entire comment.

File: test/jaxp/javax/xml/jaxp/unittest/transform/SurrogateTest1.xml (hunk 2 above)
Comment (masyano) [reply test/jaxp/javax/xml/jaxp/unittest/transform/SurrogateTest1.xml]: I added a new line at the end of every new file.

**Review by LanceAndersen — APPROVED** [review]
The updated changes look reasonable to me.

**Review by naotoj — APPROVED** [review]
Looks good. Thank you for the fix!

**Review by irisclark — APPROVED** [review]

**Comment by JoeWang-Java** [comment]
Thanks for the update. A full test passed.

**Review by JoeWang-Java — APPROVED** [review]

**Comment by JoeWang-Java** [comment]
Hi Masanori, you may now type /integrate in a new comment when you're ready. I'll then sponsor your change. Thanks.

**Pull request closed without merging**